### PR TITLE
Format selection reasons

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following starts to implement the reasons why a format was selected.
We can use that information to improve the error messages for users
to better understand.

Work towards fix the following bug: https://bugs.launchpad.net/juju/+bug/1936281

### CODE REVIEW

File: charmarchive.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func readCharmArchive(zopen zipOpener) (archive *CharmArchive, err error) {
 	// this charm is v1 or not.
 	reader, err = zipOpenFile(zipr, "manifest.yaml")
 	if _, ok := err.(*noCharmArchiveFile); ok {
-		b.manifest = NewManifest()
```
Comment (manadart) [inline charmarchive.go:84]: Do we still need `NewManifest` if it's never non-empty?

**Review by manadart — APPROVED** [review]
Looks good.

File: charmarchive.go (hunk 1 above)
Comment (SimonRichardson) [reply charmarchive.go:84]: No idea, will check.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**